src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesFactoryAdapter.java

```diff
@@ -107,11 +107,9 @@ public KubernetesClient createClient() throws KubernetesAuthException {
             LOGGER.log(FINE, "Autoconfiguring Kubernetes client");
             builder = new ConfigBuilder(Config.autoConfigure(null));
         } else {
-            // although this will still autoconfigure based on Config constructor notes
-            // In future releases (2.4.x) the public constructor will be empty.
-            // The current functionality will be provided by autoConfigure().
-            // This is a necessary change to allow us distinguish between auto configured values and builder values.
-            builder = new ConfigBuilder().withMasterUrl(serviceAddress);
+            // Using Config.empty() disables autoconfiguration when both serviceAddress and auth are set
+            builder = auth == null ? new ConfigBuilder() : new ConfigBuilder(Config.empty());
+            builder = builder.withMasterUrl(serviceAddress);
```

Comment:
Maybe add a reference to the upstream issue? So this can be reverted once fixed.
Or is this intended to be a permanent fix?

Reply:
I think we really don't want autoconfiguration when providing explicit connection details.